### ISSUE TITLE
Create a school authentication system

### ISSUE THREAD

**Issue by Dalordish** [MelbourneHighSchoolRobotics/RCJA_Registration_System#21]
Currently, anyone can sign up and become a mentor of a school, then make changes to that school.

I'm open to suggestions, but the best way I've thought of (in prior discussions with @peter-drew ) is to allow current mentors of schools to set/access a passcode that new mentors are required to enter to validate that they actually belong to the school.

Alternate ways to implement this that I think are clunkier/not as good are:

-Allow mentors to request joining a school (much like how facebook group acceptance is implemented)
-Allow mentors to send a custom invite link to join a school (like a password reset email, except for signing up)

To summarise, the user flow would look something like this:
1. Mentor 1 registers their school and makes teams
2. Mentor 2 now wants to join that school
3. Mentor 2 requires a passcode to join that school and make edits to it
4. Mentor 1 provides that code to Mentor 2.
5. Mentor 2. can now successfully register

**Comment by Dalordish** [comment]
@peter-drew let's have a quick chat about this soon and then implement

**Comment by peter-drew** [comment]
Added the key field to the db. If key is empty string, adding members disabled (could remove school from dropdown list even). Otherwise need to authenticate that the key is correct. Need to decide if key is user settable or randomly generated, leaning towards user settable, which is the current and easier implementation. Will need to add the field to the mentor facing school editing page.
@Dalordish

**Comment by peter-drew** [comment]
Need to also consider case of freshly created school. Probably allow anyone to add themselves as a mentor to a school that has no mentors and no teams.

**Comment by peter-drew** [comment]
Long term probably good to restructure the signup page in light of the changes discussed yesterday. Short term will just filter the dropdown based on schools without teams and administrators, which will capture newly created schools.

**Comment by peter-drew** [comment]
Also need ability to add mentor by email on school page
